api/src/main/java/org/openmrs/module/fhir/api/impl/PersonServiceImpl.java

```diff
@@ -92,4 +95,61 @@ public Person createFHIRPerson(Person person) {
 		return FHIRPersonUtil.generatePerson(omrsPerson);
 	}
 	
+	@Override
+	public Person updateFHIRPerson(Person thePerson,String theId){
+		org.openmrs.api.PersonService personService = Context.getPersonService();
+		org.openmrs.Person retrievedPerson=personService.getPersonByUuid(theId);
+		if(retrievedPerson!=null){ // update person
+			org.openmrs.Person omrsPerson = FHIRPersonUtil.generateOpenMRSPerson(thePerson);
+			
+			Set<PersonName> all=retrievedPerson.getNames();
+			boolean needToSetPrefferedName=false; // indicate wheter any preffered names are in the request body. 
+			for(PersonName name: omrsPerson.getNames()){
+				if(name.getPreferred()){   // detecting any preffered names are in the request body
+					needToSetPrefferedName=true;
+				}
+			}
+			if(needToSetPrefferedName){  // unset the existing preffered name, 
+				for(PersonName name: all){
+					name.setPreferred(false);
```

Comment:
no need to check whether name is already not preferred ? we only need to set false iff name is preferred